share/functions/__fish_shared_key_bindings.fish

```diff
@@ -130,16 +130,48 @@ function __fish_shared_key_bindings -d "Bindings shared between emacs and vi mod
     # Re-running `bind` multiple times per mode is still faster than trying to make the list unique,
     # even without calling `sort -u` or `uniq`, for the vi-bindings.
     # TODO: This can be solved better once #3872 is implemented.
+
+    # We usually just pass the text through as-is to facilitate pasting code,
+    # but when the current token contains an unbalanced single-quote (`'`),
+    # we escape all single-quotes and backslashes, effectively turning the paste
+    # into one literal token, to facilitate pasting non-code (e.g. markdown or git commitishes)
     set -l allmodes default
     set allmodes $allmodes (bind -a | string match -r -- '-M \w+' | string replace -- '-M ' '')
     for mode in $allmodes
-        bind -M $mode -m paste \e\[200~ 'set -g __fish_last_bind_mode $fish_bind_mode'
+        bind -M $mode -m paste \e\[200~ '
```

Comment:
I'm not a fan of a bind command literal being this long since it's hard to recognize it is a single parameter to the bind command. I wasn't going to insist abstracting this block into a private function until I noticed the next bind command at line 159 with an even longer block that needed subtle quoting and escaping.